### PR TITLE
#182631733: CUSTOM FILTER AND SEARCH

### PR DESCRIPTION
**Pull request by MwanikiN**
#### What does this Pull Request do?
. Create custom search and filter functionality
#### Description of Task to be completed?
. Create custom search and filter functionality
#### How should this be manually tested?
.Via postman using the specified search fields or text
#### Any background context you want to provide?
. Both filter and search techniques help return specific information queried by the user.
#### What are the relevant Pivotal tracker card?
[Add Search Functionality and Custom filtering based on parameters](https://www.pivotaltracker.com/n/projects/2570604)
#### Screenshots (if appropriate)
image

#### Questions:
N/A

### CODE REVIEW

**Comment by narravabrion** [comment]
Looks good

File: articles/views.py [hunk 1]
```diff
@@ -24,6 +36,16 @@ class ArticleDetailView(generics.RetrieveUpdateDestroyAPIView):
     lookup_field = "slug"
     queryset = Article.objects.all()
     renderer_classes = (JSONRenderer,)
+    filter_backends = [DjangoFilterBackend, SearchFilter]
```
Comment (mainamuriuki274) [inline articles/views.py:39]: Do you need to filter for one ?

File: core/settings/base.py [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "rest_framework_simplejwt.token_blacklist",
     "drf_yasg",
     "taggit",
+    "django_filters",
```
Comment (mainamuriuki274) [inline core/settings/base.py:35]: Where are you using this?

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: core/settings/base.py (hunk 2 above)
Comment (MwanikiN) [reply core/settings/base.py:35]: Used django_filters to implement filtering

File: articles/views.py (hunk 1 above)
Comment (MwanikiN) [reply articles/views.py:39]: I'm not clear on this one

**Pull request closed without merging**